pkg/server/server.go

```diff
@@ -370,44 +371,51 @@ func (s *ProxyServer) Proxy(stream client.ProxyService_ProxyServer) error {
 	recvCh := make(chan *client.Packet, xfrChannelSize)
 	stopCh := make(chan error)
 
-	go s.serveRecvFrontend(stream, recvCh)
+	labels := runpprof.Labels(
+		"serverID", s.serverID,
+		"serverCount", strconv.Itoa(s.serverCount),
+		"userAgent", strings.Join(userAgent, ", "),
```

Comment:
I'd expect this all to be constant for a given apiserver-proxyserver pair, but I think it would be good to be able to associate go-routines with an individual FE connection. I suggest adding a unique frontend-id. For now I'd just suggest something like this:

```suggestion
		"serverID", s.serverID,
		"serverCount", strconv.Itoa(s.serverCount),
		"userAgent", strings.Join(userAgent, ", "),
		// TODO: propagate frontendID to logging and other contexts.
		"frontendID", uuid.New().String(),
```

This relies on the stickiness to associate the frontendID with related goroutines, and we can handle the TODO later.

Reply:
Right now we do not mutex the tunnel between apiserver-proxyserver pair. So the uuid here would be more of a frontend tunnel id rather than a frontend id. Its effectively the same thing as dialID and connID today. (This should not be true) I'd rather spend time setting that up right rather than rushing something that might not be quite what we want. I believe this PR has value without that change.